### PR TITLE
Convert from project.json to csproj

### PR DESCRIPTION
**Pull request by Tratcher**
Still need to figure out the .tt files for generated code.

### CODE REVIEW

**Comment by natemcmaster** [comment]
Are .tt files a Visual Studio only thing?

**Comment by Tratcher** [comment]
Filed https://github.com/aspnet/HttpSysServer/issues/309 for followup

File: samples/SelfHostServer/SelfHostServer.csproj [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFrameworks>net451;netcoreapp1.1</TargetFrameworks>
+    <!-- TODO remove when https://github.com/dotnet/sdk/issues/396 is resolved -->
+    <RuntimeIdentifier Condition=" '$(TargetFramework)' != 'netcoreapp1.1' ">win7-x64</RuntimeIdentifier>
+    <OutputType>Exe</OutputType>
+  </PropertyGroup>
+  <ItemGroup>
+    <None Include="App.config" />
```
Comment (JunTaoLuo) [inline samples/SelfHostServer/SelfHostServer.csproj]: ?

Comment (JunTaoLuo) [inline samples/SelfHostServer/SelfHostServer.csproj]: I think the expected way to do this is to remove the `App.config` file and add a `runtimeconfig.template.json` instead. https://github.com/aspnet/WebSockets/blob/dev/samples/EchoApp/runtimeconfig.template.json

Comment (natemcmaster) [inline samples/SelfHostServer/SelfHostServer.csproj]: Don't you need both to make server GC work on .NET Framework?

Comment (JunTaoLuo) [inline samples/SelfHostServer/SelfHostServer.csproj]: Ah, I see the `runtimeconfig.template.json` is only for netcoreapp? I thought it was used by both tfms. In that case, we'll need something like `<Content Include="<filename>" CopyToOutputDirectory="PreserveNewest" CopyToPublishDirectory="PreserveNewest" />` right?

Comment (natemcmaster) [inline samples/SelfHostServer/SelfHostServer.csproj]: You don't need the content item either. The SDK is smart enough to recognize App.config and to fold it into the generated config file for .NET Framework. Likewise for runtimeconfig.template.json for .NET Core.

**Comment by Tratcher** [comment]
https://github.com/aspnet/HttpSysServer/issues/307

**Pull request merged**